src/Cake.SqlServer/SqlDacpacImpl.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using Cake.Core;
+using Cake.Core.Diagnostics;
+using Microsoft.SqlServer.Dac;
+
+namespace Cake.SqlServer
+{
+    internal static class SqlDacpacImpl
+    {
+        /// <summary>
+        /// Extract the schema from a database into a package.
+        /// </summary>
+        /// <param name="context">The context.</param>
+        /// <param name="connectionString">The connection string.</param>
+        /// <param name="targetDatabaseName">The target database.</param>
+        /// <param name="settings">The settings.</param>
+        public static void ExtractDacpacFile(ICakeContext context, string connectionString, string targetDatabaseName, ExtractDacpacSettings settings)
+        {
+            context.Log.Information($"About to extract a dacpac file from database {targetDatabaseName}");
+
+            var service = new DacServices(connectionString);
+
+            service.Extract(settings.OutputFile.FullPath, targetDatabaseName, settings.Name, settings.Version, settings.Description, settings.Tables);
+
+            context.Log.Information($"Finished creating dacpac file from database {targetDatabaseName}. File location is {settings.OutputFile}");
+        }
+
+        /// <summary>
+        /// Extract the schema from a database into a package.
+        /// </summary>
+        /// <param name="context">The context.</param>
+        /// <param name="connectionString">The connection string.</param>
+        /// <param name="targetDatabaseName">The target database.</param>
+        /// <param name="dacpacFilePath">Path to the dac file.</param>
+        /// <param name="config">Configure the sql deployment</param>
+        public static void PublishDacpacFile(ICakeContext context, string connectionString, string targetDatabaseName, string dacpacFilePath, Action<PublishOptions> config = null)
```

Comment:
@trailmax You are right, this is probably fine, I just have my concerns about the usability of the API in one command instead of having to create a setting object and initialize the Internal Options as well and set the right properties. I prefer, from the user perspective to just set the property values and voila. But is your call, what do you want?